src/aria/widgets/form/DatePicker.js

```diff
@@ -378,6 +439,111 @@ module.exports = Aria.classDefinition({
             } else {
                 this.$DropDownTextInput._onBoundPropertyChange.call(this, propertyName, newValue, oldValue);
             }
+        },
+
+        _reactToControllerReport : function (report) {
+            // --------------------------------------------------- destructuring
+
+            var cfg = this._cfg;
+            var waiAria = cfg.waiAria;
+
+            var ok = report.ok;
+            var value = report.value;
+
+            // ------------------------------------------------------ processing
+
+            if (waiAria && ok && value != null) {
+                this._waiReadDate(value);
+            }
+
+            // ------------------------------------------------------ delegation
+
+            this.$DropDownTextInput._reactToControllerReport.apply(this, arguments);
+        },
+
+        _waiReadDate : function (value) {
+            // --------------------------------------------------- destructuring
+
+            var cfg = this._cfg;
+            var waiAriaConfirmDateFormat = cfg.waiAriaConfirmDateFormat;
+
+            // ------------------------------------------------------ processing
+
+            var outputText = ariaUtilsDate.format(value, waiAriaConfirmDateFormat);
+            this.waiReadText(outputText);
         }
     }
 });
+
+
+
+////////////////////////////////////////////////////////////////////////////////
+// Helpers
+////////////////////////////////////////////////////////////////////////////////
+
+function nop() {}
+
+function Debouncer(spec) {
+    this.delay = spec.delay;
+    this.state = spec.state;
+
+    var onStart = spec.onStart;
+    if (onStart == null) {
+        onStart = nop;
+    }
+    this.onStart = onStart;
+
+    var onCall = spec.onCall;
+    if (onCall == null) {
+        onCall = nop;
+    }
+    this.onCall = onCall;
+
+    var onEnd = spec.onEnd;
+    if (onEnd == null) {
+        onEnd = nop;
+    }
+    this.onEnd = onEnd;
+
```

Comment:
Note that lines 489-507 could be replaced by:
```js
this.onStart = spec.onStart || nop;
this.onCall = spec.onCall || nop;
this.onEnd = spec.onEnd || nop;
```
3 lines instead of 18.
I personally prefer concise code, especially if it is more readable as it would be the case here.